src/core.js

```diff
@@ -1,3 +1,4 @@
+import * as d3 from 'd3';
```

Comment:
That's sort of the question I was asking; it ultimately results in D3 being provided by the bundle and not a global — as such, users don't need to include D3 separately, just C3. If you're wanting to do something clever with C3 (i.e., use it as a dependency in a larger project), you should probably be using a bundler anyway.

Another possibility would be to set D3 as an external dependency; see https://github.com/rollup/rollup/wiki/JavaScript-API#external